### DISCUSSION TITLE
Get public key from private key

### DISCUSSION THREAD

**Discussion by kor3k** [openpgpjs/openpgpjs#1283] [Q&A] [answered]
this might be a stupid question, but how can i get/derive public key from private key?

```
const privKey = await openpgp.readKey('-----BEGIN PGP PRIVATE KEY BLOCK....')
const pubKey = getPubKeyFromPrivKey(privKey)
```
is it possible? is it the one in `subKeys` prop?

**Comment by twiss** [comment, marked as answer]
Hey :wave: `privKey.toPublic()` should work (and then `.armor()` if you want it armored).

**Reply by odds-get-evened** [reply]
how do we get the serialized uint8array out of this? when using the the keypair generator, we get a serialized public key property, but not when we extract the public key from the secret on using toPublic()?

**Reply by odds-get-evened** [reply]
...and you guys have formatObject() documented but when I call this function it doesn't exist. thoroughly confusing...

**Reply by twiss** [reply]
> how do we get the serialized uint8array out of this?

`privKey.toPublic().write()`

> ...and you guys have formatObject() documented but when I call this function it doesn't exist. thoroughly confusing...

Right, that one should be marked as private. Do you want to open an issue (or even better, a PR) for that?
